## Google Playstore and iOs App Store Apps Analysist

In this analysist, we aim to build a profile of profitable apps.  The apps our developers will build will be free to download and install.  This leaves in-app ads as our profit.

We are building a profile of free profitable apps to assist our development team to know what type of apps to build.  The more users our apps get the more profit our company makes.  The type of apps we want to build are the apps the will bring in a lot of users.

In [1]:
from csv import reader

#Open up the android store data set
opened_file = open('Data_Sets\googleplaystore.csv', encoding='utf8')
read_file = reader(opened_file)
android = list(read_file)
#Split the list of lists into a header list and a data set list of lists
android_header = android[0]
android = android[1:]

#Open up the apple store data set
opened_file = open('Data_Sets\AppleStore.csv', encoding='utf8')
read_file = reader(opened_file)
iOs = list(read_file)
#Split the list of lsits into a header list and a data set list of lists
iOs_header = iOs[0]
iOs = iOs[1:]

The data quest mission provided a nice `explopre_data()` function that prints the data sets with a blank line inbetween so it is easier to read what is in the data set. It will also provide the number of rows and columns in the dataset. The rows will show how many apps there are and the columns will show what catagories the data set provides data on.

In [2]:
#The explore_data function was provided by the data quest mission
def explore_data(dataset, start, end, rows_and_columns=False):
    #grab only the lists that we want from the data set
    dataset_slice = dataset[start:end]
    #prints the lists that we grabbed with a blank line in between
    for row in dataset_slice:
        print(row)
        print('\n')
    
    #prints the number of rows and columns if that was specified
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [3]:
#Print the android header and first 5 lines of the android data set
print(android_header)
print('\n')
explore_data(android, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

With the `explore_data()` function we get a good look at the first 5 rows of the android playstore.  We also learn that there are 10841 rows in the data set.

In [4]:
#Print the ios header and first 5 lines of the ios data set
print(iOs_header)
print('\n')
explore_data(iOs, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


With the `explore_data()` function we get a good look at the first 5 rows of the iOs appstore data.  We also learn that there are 7197 rows in the data set.


## Data cleaning
Our company only wants data on free apps that are aimed towards the English-speakin market.  As part of the data cleaning process we will want to remove any paid apps and any apps that arn't in english.  There may also be duplicates and incorrect data that we need to clean up too.

The data sets come with discussion threads on each.  We will be looking at these discussion threads to see if we can identify any rows that have data we don't want.

The google playstore discussion is located [here](https://www.kaggle.com/lava18/google-play-store-apps/discussion)
The ios appstore discussion is located [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/discussion)

In [5]:
#print the header for reference against the incorrect rows
print(android_header)
print('\n')
#a row with an incorrect ratings score
print(android[10472])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [6]:
#row 10472 has a rating of 19 but the max rating is 5
#we should only run this code once.
del android[10472]

In [7]:
#the length of the android data set is now 10840 instead of 10841
print(len(android))

10840


### Data Duplicates
Now we will be looking for duplicates in the data sets.  Lets start with the google playstore data set first.  We will be creating a list of duplicate entries and a list of unique entries.  If an app appears in the unique entry then we place it in the duplicate entry.

In [8]:
#create 2 empty lsits
unique_list = []
duplicate_list = []

#loop through the android data set
for app in android:
    #grab the name of the app
    name = app[0]
    #if the app is in the unique list then place it in the duplicate list
    if name in unique_list:
        duplicate_list.append(name)
    #otherwise place it in the unique list
    else:
        unique_list.append(name)
        
#then we want to see how many duplicates there are and some examples
print(len(duplicate_list))
print('\n')
print(duplicate_list[:10])

1181


['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


In [9]:
#'Slack' is a duplicate so lets look at the rows in the data set with name 'Slack'
#print the header for reference
print(android_header)
print('\n')

#loop through the android data set
for app in android:
    #grab the name of the app
    name = app[0]
    #if the app name is 'Slack' print it
    if name == 'Slack':
        print(app)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51510', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']



We have 1181 duplicate rows.  We have to delete these rows before we can analyze the data otherwise our data would not be as accurate as we want.  As we can see when we printed the `Slack` duplicate rows in the data set, the reviews is not the same on all 3.  The app with the highest number of reviews will have the most accurate rating.  So instead of deleteing the duplicates at random, we will instead delete the duplicates with the lowest number of reviews.  


In [10]:
#create an empty dictionary
reviews_max = {}

#loop through the google dataset
for app in android:
    #grab the name and the review
    name = app[0]
    #set the reviews as float
    num_reviews = float(app[3])
    
    #check if name exists in reviews_max and num_reviews is greater
    if name in reviews_max and num_reviews > reviews_max[name]:
            #then update reviews_max
            reviews_max[name] = num_reviews
    #if name doesn't exist in reviews_max then add it
    elif name not in reviews_max:
        reviews_max[name] = num_reviews

#check if the dictionary is the length we expect
#the android data set minus 1181 duplicates is what we expect
print('Expected Lenght: ', len(android) - 1181)
print('New Length: ', len(reviews_max))

Expected Lenght:  9659
New Length:  9659


In the above cell we created a dictionary which contained a full list of app names from the google data set.  In this dictionary we stored the name as the `key` and the number of reviews as the `value`.  We also made sure to remove all of the duplicates so the dictionary only contained unique apps and the highest number of reviews per app.

In the next cell we will use this dictionary to create a new clean list of adroid data. To create the list we will:
1. create 2 empty lists which are `android_clean` and `already_added`
2. loop thought the google data set for each loop
    1. we grab the `name` and `num_reviews` same as we did in the previous cell
    2. compare the `name` and `num_reviews` to `review_max`
        1. we use `review_max` to ensure that the app we add is the app with the highest number of reviews
    3. if the app is the one with the highest `num_reviews` then
        1. we add the app to `android_clean`
        2. we add the name to `already_added`
            1. we use `already_added` for the cases where we have duplicates that have the same number of reviews
            
This will give us a list `android_clean` which will have no duplicates in it.

In [11]:
#create 2 empty lists
android_clean = []
already_added = []

#loop through the google dataset
for app in android:
    #get the name and review number as a float
    name = app[0]
    num_reviews = float(app[3])
    
    #if the number of reviews is the one in reviews_max and name isn't in already_added
    if num_reviews == reviews_max[name] and name not in already_added:
        #append the full app to android_clean
        android_clean.append(app)
        #append the name to already_added
        already_added.append(name)
        
#check the length of android_clean with explore_data
print(android_header)
print('\n')
explore_data(android_clean, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Every

Now we can do the same duplicate check on the iOs data. We find that there are no duplicates in the iOs data set so we don't have to remove any.

In [12]:
#create 2 empty lsits
unique_list = []
duplicate_list = []

#loop through the iOs data set
for app in iOs:
    #grab the name of the app
    name = app[0]
    #if the app is in the unique list then place it in the duplicate list
    if name in unique_list:
        duplicate_list.append(name)
    #otherwise place it in the unique list
    else:
        unique_list.append(name)
        
#then we want to see how many duplicates there are
print(len(duplicate_list))

0


### Removing Non-English Apps
Our company will be building apps directed toward English Speaking users.  As such the next thing we want to do for data cleaning is removing any non-English apps.  

In [13]:
#create a function that looks at a string
#returns False if the string contains non-English characters
#returns True if the string is in all English
def check_english(arg_string):
    #create a counter for counting non-English characters
    counter = 0
    
    #loop through the string
    for i in arg_string:
        #English characters have ASCII values of 0 through 127
        #check if the character is greater than 127
        if ord(i) > 127:
            #then add to the counter
            counter += 1
            
    #outside of the loop we have a final counter
    #if the counter is greater than 3 then return False
    if counter > 3:
        return False
    #else return True
    else:
        return True

#check if the function works
print(check_english('Instagram'))
print(check_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(check_english('Docs To Go™ Free Office Suite'))
print(check_english('Instachat 😜'))

True
False
True
True


In thew above cell we create a function to check for English character.  As English characters make up the ASCII values 0-127 we use that to check.  But we run into the problem of the TM symbol and the emoji symbol being outside of the range of 0-127.  To solve then we create a counter and if there are more than 3 symbols outside of the range then we return `False`. Otherwise, we return `True`

After that we run the 2 data sets we are using throught the function.  For the android data set we use the clean data set we made by removing duplicates.  For the iOs data set we had no duplicates so we can use the original data set.

In [14]:
#create two new lists for the data sets
android_english = []
iOs_english = []

#loop through android_clean
for app in android_clean:
    #grab the name of the app
    name = app[0]
    #check if the name is English
    if check_english(name):
        #then add the app to the new English list
        android_english.append(app)

#loop through iOs
for app in iOs:
    #grab the name of the app
    name = app[1]
    #check if the name is English
    if check_english(name):
        #then add the app to the new English list
        iOs_english.append(app)
        
#explore the lists to ensure it works
print(android_header)
print('\n')
explore_data(android_english, 0, 5, True)
print('\n')
print(iOs_header)
print('\n')
explore_data(iOs_english, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Every

### Grabbing free apps
Now that our data is free of duplicates and non-English apps, we can start grabbing the apps that we want.  Our company only makes free apps so we will start by going through the data sets and removing all the paid apps.

In [15]:
#create 2 new lists
android_free = []
iOs_free = []

#loop through the android_english data set
for app in android_english:
    #the price catagory for the android data set is the 7th index
    price = app[7]
    #check if the price is 0
    if price == '0':
        #then add it to android_free
        android_free.append(app)
        
#loop through the iOs_english data set
for app in iOs_english:
    #the price catagory for the iOs data set is the 4th index
    price = app[4]
    #check if the price is 0.0
    if price == '0.0':
        #then add it to iOs_free
        iOs_free.append(app)
        
#now we can explore explore the data one more time
print(android_header)
print('\n')
explore_data(android_free, 0, 5, True)
print('\n')
print(iOs_header)
print('\n')
explore_data(iOs_free, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Every

The total number of free apps we are left with is 8864 apps from the android playstore and 3222 apps from the iOs app store.

## Most Common Apps by Genre
We want to develope apps that want to reach the most amount of users we can.  Our strategy is going be 3 steps:
1. We build an Android app and add it to the android playstore.
2. If the app gets a good response then we continue developing it
3. After six months, if the app is profitable then we build an iOs version and add it to the app store

Because at the end of the 3 steps we will have an app on both the android playstore and the iOs appstore.  We want to look at apps that will perform well on both stores.  As part of that we will be analyzing the data of both data sets.

The catagories we will be looking at are the genre catagories.  For the android play store we have `catagory` which is index `1` and `genre` which is index `9`.  For the iOs app store we have `prime_genre` which is index `11`

In [16]:
#create a function that makes a frequency table
#takes in 2 arguments:
#   dataset is the list of lists the table will be of
#   index is which list in the list of lists the table will be of
def freq_table(dataset, index):
    #creat an empty dictionary for the frequecy table
    freq_table = {}

    #loop through the data set
    for row in dataset:
        #grab the value of the row
        value = row[index]
        #check if the value is in the dictionary
        if value in freq_table:
            #add one to the value
            freq_table[value] += 1
        #else create a new dictionary entry
        else:
            #set the new entry to 1
            freq_table[value] = 1
    
    #now to turn the frequncies to a percentage
    #create an empty dictionary for percentages
    percent_table = {}
    
    #calculate the length for the percentage table
    length = len(dataset)
    #loop through the frequency table
    for i in freq_table:
        #calculate the length of the frequency table
        #calculate the percentage
        percent = (freq_table[i] / length) * 100
        #add the percentage to the percent_table
        percent_table[i] = percent
        
    #return the percentage table
    return percent_table

In [17]:
#The display_table function was provided by the data quest mission
def display_table(dataset, index):
    #get a percentage table
    table = freq_table(dataset, index)
    #create an empty list for displaying
    table_display = []
    
    #loop though the percentage table
    for key in table:
        #set the value, key as a tuple
        key_val_as_tuple = (table[key], key)
        #add the tuple to the display table
        table_display.append(key_val_as_tuple)

    #sort the table in descending order
    table_sorted = sorted(table_display, reverse = True)
    #loop through the sorted table
    for entry in table_sorted:
        #print the entries
        print(entry[1], ':', entry[0])

Now that we have a functions `freq_table` which takes in the data set and an index and returns a percentage dictionary for the data set. and a function `display_table` which takes in the percentage table, sorts it into descending order, and then prints it out in a nice list.  We can start printing out the catagories that we want.
The catagories that we want are `catagory` and `genre` from the android playstore and `prime_genre` from the iOs appstore

In [18]:
#print the catagory for google playstore using the functions
display_table(android_free, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In this table we look at `catagory` in the google playstore data set.  What we see is that Family is the highest catagory with 18.90% followed by Game at 9.72%.  If we continue down the list we see that utility apps such as Family, Tools, Business, Lifestyle, and productivity make up most of the top catagories.  While fun apps such as Games, Sports, and Photography have much less representation.

The general trend seems to be on utility over fun.  While Games still has 2nd place it is the only fun app catagory we see until we get down to Sports in 9th place.

This app profile is limited to just what we were looking at which is free English games so it might not represent the appstore overall.  This profile also only looks at the number of apps per genre and not user rating or even how many users each genre gets.  Thus, the percentage of each genre does not necessarily show user demand.

In [19]:
#print the genre for google playstore using the functions
display_table(android_free, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In this table we look at `genre` in the google playstore.  One thing to note is that `genre` and `catagory` don't seem to be too different.  `Genre` does have more of a breakdown than `catagory` which for us is not too important right now.  What was said about `catagories` can then be applied to `genre` in terms of general trend and utility over fun.

In [20]:
#print the prime_genre for apple appstore using the functions
display_table(iOs_free, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In this table for the `prime_genre` catagory of the apple appstore we get a nice list of percentages.  We find that the most common genre is Games followed by Entertainment and Photo & Video.  The Games genre take a majority of the apps at just over 58%, with Entertainment dropping down to only 7.88%.

A general trend we can see is that fun and entertainment are the highest trends.  Games, Entertainment, Photo & Video, Social Media are all within the top 5 catagories.  Education does beat Social Media by about half a percentage but still no where near as much of a percentage as Games.

This app profile is limited to just what we were looking at which is free English games so it might not represent the appstore overall.  This profile also only looks at the number of apps per genre and not user rating or even how many users each genre gets.  Thus, the percentage of each genre does not necessarily show user demand.

The differences between google playstore and apple appstore are quite intresting.  The android store seems to have a lot more apps based on utility over fun.  While the apple store is the opposite and has a lot more apps based on fun over utility.  Then we look at the percentages and see that the playstore is a lot more balanced than the appstore, with the highest playstore percentage being 18.90% vs. the appstore 58.16%.

### Most Popular Apps by Genre
Now that we have good lists of the common apps by genre, Lets look at the same genre catagories except we will take into account how many installs the apps have aswell.

In terms of the google playstore we can use the `installs` section which is index `5`.  For the apple appstore there is no `installs` section so instead we can look at the total rating count `rating_count_tot` which is index `5`.

##### Popular Apps on the iOs Appstore
First lets look at the iOs store and see which genre has the best rating counts compared to the number of apps.

In [42]:
#create a dictionary of unique genres in the iOs store
unique_genre = freq_table(iOs_free, 11)

#loop over the unique_genre list
for genre in unique_genre:
    #create a total variable
    total = 0
    #create a genre counter variable
    len_genre = 0
    
    #loop over the iOs dataset
    for app in iOs_free:
        #get the app_genre
        genre_app = app[11]
        
        #if the genre is the same as genre_app
        if genre == genre_app:
            #save the user rating as a float
            user_rating = float(app[5])
            #add the user_ratings to total
            total += user_rating
            #increment the genre counter
            len_genre += 1
            
    #get the average
    average = total / len_genre
    #print the genre : average
    print("{} : {:,.4f}".format(genre, average))

Social Networking : 71,548.3491
Photo & Video : 28,441.5438
Games : 22,788.6697
Music : 57,326.5303
Reference : 74,942.1111
Health & Fitness : 23,298.0154
Weather : 52,279.8929
Utilities : 18,684.4568
Travel : 28,243.8000
Shopping : 26,919.6905
News : 21,248.0233
Navigation : 86,090.3333
Lifestyle : 16,485.7647
Entertainment : 14,029.8307
Food & Drink : 33,333.9231
Sports : 23,008.8986
Book : 39,758.5000
Finance : 31,467.9444
Education : 7,003.9831
Productivity : 21,028.4107
Business : 7,491.1176
Catalogs : 4,004.0000
Medical : 612.0000


The top 5 genres are Navigation, Reference, Social Networking, Music, and Weather.  If we look at the top 5 genres and what apps there are in each genre, we could get a clearer picture as to why they are the top 5 genres. 

In [28]:
#loop through the iOs data set
for app in iOs_free:
    #check if the genre is Navigation
    if app[11] == 'Navigation':
        #print the app name and ratings
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


As we see there are only 6 navigations apps.  With Waze having the majority of ratings and Google Maps in second.  This genre won't do us any good because anything besides Waze and Google Maps aren't getting good numbers.

In [29]:
#loop through the iOs data set
for app in iOs_free:
    #check if the genre is Reference
    if app[11] == 'Reference':
        #print the app name and ratings
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


For the Reference genre, we see more balanced review numbers than the Navigation genre.  This could be a good contender for the type of app we want to make.

In [30]:
#loop through the iOs data set
for app in iOs_free:
    #check if the genre is Social Networking
    if app[11] == 'Social Networking':
        #print the app name and ratings
        print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In the Social Networking genre, once again we see a skew to the top of reviews.  With Facebook at the very top almost tripling the next highest ratings app.  Then we look towards the bottom of the list and see quite a number of apps  below 10,000 reviews.  This genre seems like a hard genre to break into so we can ignore this one.

In [31]:
#loop through the iOs data set
for app in iOs_free:
    #check if the genre is Music
    if app[11] == 'Music':
        #print the app name and ratings
        print(app[1], ':', app[5])

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

Next up is the Music genre.  Again, we see a few apps dominating the ratings while most apps are low on the number of ratings.  Another bad genre to break into.

In [32]:
#loop through the iOs data set
for app in iOs_free:
    #check if the genre is Weather
    if app[11] == 'Weather':
        #print the app name and ratings
        print(app[1], ':', app[5])

The Weather Channel: Forecast, Radar & Alerts : 495626
The Weather Channel App for iPad – best local forecast, radar map, and storm tracking : 208648
WeatherBug - Local Weather, Radar, Maps, Alerts : 188583
MyRadar NOAA Weather Radar Forecast : 150158
AccuWeather - Weather for Life : 144214
Yahoo Weather : 112603
Weather Underground: Custom Forecast & Local Radar : 49192
NOAA Weather Radar - Weather Forecast & HD Radar : 45696
Weather Live Free - Weather Forecast & Alerts : 35702
Storm Radar : 22792
QuakeFeed Earthquake Map, Alerts, and News : 6081
Moji Weather - Free Weather Forecast : 2333
Hurricane by American Red Cross : 1158
Forecast Bar : 375
Hurricane Tracker WESH 2 Orlando, Central Florida : 203
FEMA : 128
iWeather - World weather forecast : 80
Weather - Radar - Storm with Morecast App : 78
Yurekuru Call : 53
Weather & Radar : 37
WRAL Weather Alert : 25
Météo-France : 24
JaxReady : 22
Freddy the Frogcaster's Weather Station : 14
Almanac Long-Range Weather Forecast : 12
TodayAir

Finally, lets look at the Weather genre.  The Weather Channel seems to be the big app in the genre.  But this genre does seem to also be one of the more balanced genres in the top 5 genres.  This one could also be a contender for our app.

After looking at the top 5 genres with the highest number of reviews.  It would seem Reference and Weather is the most balanced genres.  In the app markets especially, building up a good catalog of apps is important instead of relying on a one hit wonder.  So perhaps building a Reference app could be our first step followed by a Weather app.

##### Popular Apps on Google Playstore
Now lets look at the Google Playstore and see which genres have the most installs compared to the number of apps.

One thing to note is that we will only be looking at `catagory` because that list was more general and we are not trying to be too specific right now.

In [43]:
#create a dictionary of uinique catagories in the android store
unique_catagory = freq_table(android_free, 1)

#loop over the catagory list
for catagory in unique_catagory:
    #create a total variable
    total = 0
    #create a catagory counter variable
    len_catagory = 0
    
    #loop over the android dataset
    for app in android_free:
        #get the app catagory
        catagory_app = app[1]

        #if the catagory is the same as catagory_app
        if catagory == catagory_app:
            #grab the number of installs
            installs = app[5]
            #clean up the number of installs by removeing ',' and '+'
            installs = installs.replace(',', '')
            installs = installs.replace('+', '')
            #set installs as a float
            installs = float(installs)
            
            #add the installs to total
            total += installs
            #increment the catagory counter
            len_catagory += 1
            
    #get the average
    average = total / len_catagory
    #print the catagory : average
    print("{} : {:,.4f}".format(catagory, average))

ART_AND_DESIGN : 1,986,335.0877
AUTO_AND_VEHICLES : 647,317.8171
BEAUTY : 513,151.8868
BOOKS_AND_REFERENCE : 8,767,811.8947
BUSINESS : 1,712,290.1474
COMICS : 817,657.2727
COMMUNICATION : 38,456,119.1672
DATING : 854,028.8303
EDUCATION : 1,833,495.1456
ENTERTAINMENT : 11,640,705.8824
EVENTS : 253,542.2222
FINANCE : 1,387,692.4756
FOOD_AND_DRINK : 1,924,897.7364
HEALTH_AND_FITNESS : 4,188,821.9853
HOUSE_AND_HOME : 1,331,540.5616
LIBRARIES_AND_DEMO : 638,503.7349
LIFESTYLE : 1,437,816.2688
GAME : 15,588,015.6032
FAMILY : 3,695,641.8198
MEDICAL : 120,550.6198
SOCIAL : 23,253,652.1271
SHOPPING : 7,036,877.3116
PHOTOGRAPHY : 17,840,110.4023
SPORTS : 3,638,640.1429
TRAVEL_AND_LOCAL : 13,984,077.7101
TOOLS : 10,801,391.2987
PERSONALIZATION : 5,201,482.6122
PRODUCTIVITY : 16,787,331.3449
PARENTING : 542,603.6207
WEATHER : 5,074,486.1972
VIDEO_PLAYERS : 24,727,872.4528
NEWS_AND_MAGAZINES : 9,549,178.4677
MAPS_AND_NAVIGATION : 4,056,941.7742


Once again we can look at the top 5 catgories.  This will include Communication, Video Players, Social, Photography, and Productivity.

In [97]:
#loop through the android data set
for app in android_free:
    #if the app is in catagory COMMUNICATION
    if app[1] == 'COMMUNICATION':
        #if the app has more than 100m installs
        if app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+':
            #print the list
            print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

We see that 6 Communications apps have over 1,000,000,000 installs.  Those apps include WhatsApp, Messanger, Skype, Google Chrome, Gmail, and Hangouts.  These apps are all owned by Facebook, Microsoft, and Google.  After that we only have 4 with 5,000,000+, then the rest fall in the 100,000,000+ range.

In [95]:
#loop through the android data set
for app in android_free:
    #if the app is in Cataogry VIDEO_PLAYERS
    if app[1] == 'VIDEO_PLAYERS':
        #if the app has more than 100m installs
        if app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+':
            #print the list
            print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


In this list of Video Players we see only 9 apps that are above 100,000,000+.  With 2 of the apps being 1,000,000,000+ and only 1 being 500,000,000+.  

In [96]:
#loop through the android data set
for app in android_free:
    #if the app is in Cataogry SOCIAL
    if app[1] == 'SOCIAL':
        #if the app has more than 100m installs
        if app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+':
            #print the list
            print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


For Social apps we see that 2 apps have 1,000,000,000+ and 2 apps have 500,000,000+.

In [98]:
#loop through the android data set
for app in android_free:
    #if the app is in Cataogry PHOTOGRAPHY
    if app[1] == 'PHOTOGRAPHY':
        #if the app has more than 100m installs
        if app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+':
            #print the list
            print(app[0], ':', app[5])

B612 - Beauty & Filter Camera : 100,000,000+
YouCam Makeup - Magic Selfie Makeovers : 100,000,000+
Sweet Selfie - selfie camera, beauty cam, photo edit : 100,000,000+
Google Photos : 1,000,000,000+
Retrica : 100,000,000+
Photo Editor Pro : 100,000,000+
BeautyPlus - Easy Photo Editor & Selfie Camera : 100,000,000+
PicsArt Photo Studio: Collage Maker & Pic Editor : 100,000,000+
Photo Collage Editor : 100,000,000+
Z Camera - Photo Editor, Beauty Selfie, Collage : 100,000,000+
PhotoGrid: Video & Pic Collage Maker, Photo Editor : 100,000,000+
Candy Camera - selfie, beauty camera, photo editor : 100,000,000+
YouCam Perfect - Selfie Photo Editor : 100,000,000+
Camera360: Selfie Photo Editor with Funny Sticker : 100,000,000+
S Photo Editor - Collage Maker , Photo Collage : 100,000,000+
AR effect : 100,000,000+
Cymera Camera- Photo Editor, Filter,Collage,Layout : 100,000,000+
LINE Camera - Photo editor : 100,000,000+
Photo Editor Collage Maker Pro : 100,000,000+


Photography apps seem to be the most balanced with every app we are looking at only being 100,000,000+ except for Google Photos.

In [99]:
#loop through the android data set
for app in android_free:
    #if the app is in Cataogry PRODUCTIVITY
    if app[1] == 'PRODUCTIVITY':
        #if the app has more than 100m installs
        if app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+':
            #print the list
            print(app[0], ':', app[5])

Microsoft Word : 500,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneDrive : 100,000,000+
Microsoft OneNote : 100,000,000+
Google Keep : 100,000,000+
ES File Explorer File Manager : 100,000,000+
Dropbox : 500,000,000+
Google Docs : 100,000,000+
Microsoft PowerPoint : 100,000,000+
Samsung Notes : 100,000,000+
SwiftKey Keyboard : 100,000,000+
Google Drive : 1,000,000,000+
Adobe Acrobat Reader : 100,000,000+
Google Sheets : 100,000,000+
Microsoft Excel : 100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
Google Slides : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
Google Calendar : 500,000,000+
Cloud Print : 500,000,000+
CamScanner - Phone PDF Creator : 100,000,000+


In Productivity, once again we see the only 1,000,000,000+ app being from Google.  Followed by a few 500,000,000+ from Microsoft word, Dropbox, Google calander.

After looking at the apps from the top 5 catagories in the google playstore that have 100 million+ installs, we can see that each catagory is dominated by one or two apps having 1 billion+ installs aside from Communications.  Then a few apps with 500 million+ also showing up from large companies in each catagory except for Photography which doesn't have any 500 million+ apps.  Photography also has a decent number of apps in the 100 million+ range despite only having one 1 billion+ and no 500 million+ apps.

Since our company's 3 step mission starts off with android then moves into iOs we can look at how the iOs Photography apps.  

In [100]:
#loop through the iOs data set
for app in iOs_free:
    #check if the genre is Photo & Video
    if app[11] == 'Photo & Video':
        #print the app name and ratings
        print(app[1], ':', app[5])

Instagram : 2161558
Snapchat : 323905
YouTube - Watch Videos, Music, and Live Streams : 278166
Pic Collage - Picture Editor & Photo Collage Maker : 123433
Funimate video editor: add cool effects to videos : 123268
musical.ly - your video social network : 105429
Photo Collage Maker & Photo Editor - Live Collage : 93781
Vine Camera : 90355
Google Photos - unlimited photo and video storage : 88742
Flipagram : 79905
Mixgram - Picture Collage Maker - Pic Photo Editor : 54282
Shutterfly: Prints, Photo Books, Cards Made Easy : 51427
Pic Jointer – Photo Collage, Camera Effects Editor : 51330
Color Pop Effects - Photo Editor & Picture Editing : 45320
Photo Grid - photo collage maker & photo editor : 40531
iSwap Faces LITE : 39722
MOLDIV - Photo Editor, Collage & Beauty Camera : 39501
Photo Editor by Aviary : 39501
Photo Lab: Picture Editor, effects & fun face app : 34585
Rookie Cam - Photo Editor & Filter Camera : 33921
FotoRus -Camera & Photo Editor & Pic Collage Maker : 32558
PicsArt Photo St

We see that Instagram is the top app with over 2 million user reviews.  With 5 apps being over 100 thousand user reviews and a good number being over 10 thousand reviews.  As user reviews are not the same as installs we can consider the review counts to be a fair amount lower than number of installs.

With this we have an idea of photography apps being a solid choice in both the google playstore and apple appstore.  We can also see that in both the playstore and appstore, a number of photography apps involve photo editing or photo effects.  Photo effects are something that is limited by creativity, so we can have everyone on the development team and even people from other teams contribute ideas.  If we create an app that provides a good variety of photo effects and even some photo editing then we could be a competing app in both the android and iOs market.

## Conclusions
In this project, our goal was to provide the development team with a direction on what type of apps to create that will reach as many users in both the google playstore and the apple appstore.  To do this we analyzed data sets that contained a large number of apps from both stores.

The conclusion we have come to is that a Photography app is a good direction to start in.  Photography has the 4th highest number of installs in the google playstore and the 7th highest number of user reviews in the apple appstore.  A good number of them are photography apps that provide photo editing and photo effects.  So to make our app more appealing, we should provide a number of photo effects and some photo editing features.  Photo effects are ideas that everyone in our company can contribue to, not just the development team.